File: app/src/main/java/de/cineaste/android/controllFlow/BaseSnackBar.java
package de.cineaste.android.controllFlow;

import android.support.v7.widget.LinearLayoutManager;
import android.view.View;

public abstract class BaseSnackBar {

    protected final LinearLayoutManager linearLayoutManager;
    protected final View view;

    protected BaseSnackBar(LinearLayoutManager linearLayoutManager, View view) {
        this.linearLayoutManager = linearLayoutManager;
        this.view = view;
    }

    public abstract void getSnackBarLeftSwipe(final int position);
    public abstract void getSnackBarRightSwipe(final int position, final int message);
}

File: app/src/main/kotlin/de/cineaste/android/controllFlow/BaseSnackBar.kt
package de.cineaste.android.controllFlow

import androidx.recyclerview.widget.LinearLayoutManager
import android.view.View

abstract class BaseSnackBar protected constructor(
    protected val linearLayoutManager: LinearLayoutManager,
    protected val view: View
) {

    abstract fun getSnackBarLeftSwipe(position: Int)
    abstract fun getSnackBarRightSwipe(position: Int, message: Int)
}
